### ISSUE TITLE
ERR Protocol error: expected '$', got '+' during tests

### ISSUE THREAD

**Issue by bre7** [vapor/redis#107]
I've been trying to run the tests in a server that requires auth (Heroku specifically) but couldn't get them to pass:

> [RedisError.serverSide: ERR Protocol error: expected '$', got '+'] [RedisDataDecoder.swift:50:94]

Could it be related to #105  ?

The only thing I modified was the `makeTest` function:

```swift
static func makeTest() throws -> RedisClient {
    let group = MultiThreadedEventLoopGroup(numThreads: 1)
    let db = try RedisDatabase(url: URL(string: "redis://h:password_here@ec2-000.compute-1.amazonaws.com:53439")!)
    return try db.makeConnection(on: group).wait()
}
```

### Repro:

1. Clone the project
2. Go to Heroku and add any (or both) of the Redis Services mentioned below
3. Copy their connection strings
4. Replace `makeTest()` with the code above, adding the proper connection string
5. Run the tests (they won't finish)

### Expected:

Tests should pass

### Actual results:

Tests will freeze / not pass

Env:
> Vapor Redis: 3.0.0-rc.2.4
> OS: macOS High Sierra 10.13.4
> Redis Services: [Heroku Redis](https://elements.heroku.com/addons/heroku-redis) & [Redis To Go](https://elements.heroku.com/addons/redistogo)

**Comment by pedantix** [comment]
this looks like  a bug related to the password issue

**Comment by bre7** [comment]
Closed in favor of #105
